logcap/client.go

```diff
@@ -38,10 +39,14 @@ func (c *Client) Ingest(ctx context.Context, batch *logsv1.IngestBatch) error {
 	}))
 	if err != nil {
 		log.Error(err, "Ingest RPC failed")
-		return err
+		var d time.Duration
+		if resp != nil && resp.Msg != nil {
+			d = resp.Msg.GetBackoff().GetDuration().AsDuration()
```

Comment:
Is it actually possible to have a response message here if we got an error?

Reply:
I'm not sure, tbh. Our (basic) test cases return a nil `resp`. If there _was_ a chance of returning an error along with a specified backoff, it felt like it would be worth returning 🤷‍♂️.

Reply:
This is the guts of the client method - it always returns a `nil` response when there is an error.

```go
	client.callUnary = func(ctx context.Context, request *Request[Req]) (*Response[Res], error) {
		// To make the specification, peer, and RPC headers visible to the full
		// interceptor chain (as though they were supplied by the caller), we'll
		// add them here.
		request.spec = unarySpec
		request.peer = client.protocolClient.Peer()
		protocolClient.WriteRequestHeader(StreamTypeUnary, request.Header())
		response, err := unaryFunc(ctx, request)
		if err != nil {
			return nil, err
		}
		typed, ok := response.(*Response[Res])
		if !ok {
			return nil, errorf(CodeInternal, "unexpected client response type %T", response)
		}
		return typed, nil
	}
```